### PR TITLE
feat: rename key

### LINKED ISSUE #454: Update key pair / rename
## Story

**As a** user
**I want** to rename my key 
**So that** I can organise my keys in the way that suits my mental model of how I use the keys

## Acceptance Criteria

- [ ] There is a way to rename a key 
- [ ] I can see the new name in the wallet UI
- [ ] The new name of a key is shared with connected dapps (backend)

## Tasks

- [ ] Detailed sketch
- [ ] Visual design
- [ ] Frontend tasks:
    - [ ] Implement UI
    - [ ] Integrate with backend
    - [ ] Write unit tests
- [ ] Backend tasks:
    - [ ] TBC
    - [ ] Write unit tests
- [ ] Write end to end tests
- [ ] Code review
- [ ] Team verify acceptance criteria met
- [ ] Exploratory QA testing 
- [ ] Capture bugs and new edge cases in tickets (this one or new TBD) 
- [ ] Write automated regression tests for the A/Cs (expanding on A/Cs where possible) 
- [ ] Add story and A/Cs to spec doc 
- [ ] Review with stakeholders
- [ ] Release

## Sketch

Image

## Notes
Rename after you've used Vega... not onboarding

Image

### PR DESCRIPTION
**Pull request by emilbayes**
Closes #454

### CODE REVIEW

File: web-extension/test/admin-ns.spec.js [hunk 1]
```diff
@@ -572,4 +591,75 @@ describe('admin-ns', () => {
...
+
+      key = generateKey.result
+    })
+
+    afterEach(() => {
+      admin = null
+      key = null
+    })
+
+    it('should not rename key with wrong public key', async () => {
```
[the remark this answers is not in the document]
Comment (dalebennett1992) [reply web-extension/test/admin-ns.spec.js:619]: also are there any special characters or things that should not be allowed in the key naming? Or are we keeping validation loose on this.

[the remark this answers is not in the document]
Comment (emilbayes) [reply web-extension/test/admin-ns.spec.js:619]: Added tests for this :+1:

Consensus seems to be that since the storage doesn't care what name you give your keys, we will keep validation loose. Right now this endpoint is also only available to the user themselves

**Comment by dalebennett1992** [comment]
image

I think 'Rename key' should be title case. So 'Rename Key'

I also is having the VegaKey component in this view a little redundant?? Would it be nicer/ simpler to just have rename key and the key field? Sorry that thought has only just occurred even though I have seen this before. Doesn't feel like it adds too much?

**Comment by dalebennett1992** [comment]
image

We probably need to address this??? When did this happen haha

File: specs/1125-KEYD-key_details.md [hunk 2]
```diff
@@ -8,3 +8,14 @@
...
 - In the key details screen I can see my currently selected [key and associated info](./1126-VKEY-vega_key.md) (<a name="1125-KEYD-007" href="#1125-KEYD-007">1125-KEYD-007</a>)
 - There is a way to export a private key (<a name="1125-KEYD-008" href="#1125-KEYD-008">1125-KEYD-008</a>)
+
+## Rename key
+
+- I can see a button to rename the key (<a name="1125-KEYD-009" href="#1125-KEYD-009">1125-KEYD-009</a>)
+- I can see the [vega key and associated info](./1126-VKEY-vega_key.md) of the key I am renaming (<a name="1125-KEYD-010" href="#1125-KEYD-010">1125-KEYD-010</a>)
+- I can input the new name (<a name="1125-KEYD-011" href="#1125-KEYD-011">1125-KEYD-011</a>)
+- When I press submit the dialog closes (<a name="1125-KEYD-012" href="#1125-KEYD-012">1125-KEYD-012</a>)
+- Submit button is disabled when key name is not entered (<a name="1125-KEYD-013" href="#1125-KEYD-013">1125-KEYD-013</a>)
```
Comment (dalebennett1992) [inline specs/1125-KEYD-key_details.md:18]: This AC ref is missing in the tests

Comment (dalebennett1992) [reply specs/1125-KEYD-key_details.md:18]: no it isnt

**Comment by dalebennett1992** [comment]
image

This issue still exists - can we fix here?

**Review by dalebennett1992 — APPROVED** [review]
looks good, backend flake keeps cropping up. Might be worth taking a look on another PR or something @emilbayes
